viscy/light/predict_writer.py

```diff
@@ -57,13 +59,33 @@ class HCSPredictionWriter(BasePredictionWriter):
 
     def __init__(
         self,
+        metadata_store: str,
         output_store: str,
         write_input: bool = False,
         write_interval: Literal["batch", "epoch", "batch_and_epoch"] = "batch",
     ) -> None:
         super().__init__(write_interval)
         self.output_store = output_store
         self.write_input = write_input
+        self.metadata_store = metadata_store
+        self._dataset_scale = None
+        self._get_scale_metadata(metadata_store)
+
+    def _get_scale_metadata(self, metadata_store: str) -> None:
+        # Update the scale metadata
+        try:
+            with open_ome_zarr(metadata_store, mode="r") as meta_plate:
+                for _, pos in meta_plate.positions():
+                    self._dataset_scale = [
+                        TransformationMeta(type="scale", scale=pos.scale)
+                    ]
+                    _logger.debug(f"Dataset scale {pos.scale}.")
+                    break
+        except IOError:
```

Comment:
When is this triggered?

Reply:
I'll swap this since we changed the input `metadata_store=None` as default behaviour. This was getting triggered if you passed anything other than a zarr.